build_runner_core/lib/src/generate/build_impl.dart

```diff
@@ -621,6 +653,53 @@ class _SingleBuild {
     }));
   }
 
+  Future<GlobAssetNode> _getUpdatedGlobNode(
+      Glob glob, String package, int phaseNum) {
+    var globNodeId = new AssetId(
+        package, 'glob.$phaseNum.${base64.encode(utf8.encode(glob.pattern))}');
+    var globNode = _assetGraph.get(globNodeId) as GlobAssetNode;
+    if (globNode == null) {
+      globNode = new GlobAssetNode(
+          globNodeId, glob, phaseNum, NodeState.definitelyNeedsUpdate);
+      _assetGraph.add(globNode);
+    }
+
+    return toFuture(
+        doAfter(_updateGlobNodeIfNecessary(globNode), (_) => globNode));
+  }
+
+  FutureOr<Null> _updateGlobNodeIfNecessary(GlobAssetNode globNode) {
```

Comment:
`FutureOr<void>` (we'll need to start fixing this elsewhere, but shouldn't introduce new cases)

Reply:
done